### PR TITLE
Comment: collapse long replies

### LINKED ISSUE #5079: Collapse long threads (x more)
On long comment threads there should be a break in the middle with "X more comments…" that you can click to expand.

### PR DESCRIPTION
**Pull request by infinite-persistence**
## Ticket
Closes #5079

## Review
- For the case of RTL, I followed how "Reply" is implemented (assumed that is the desired way). If it needs to be re-aligned, it can be fixed together with "Reply" later.
- The threshold number can be moved to `constants.ts` if we don't want to pollute the props.

### CODE REVIEW

**Comment by infinite-persistence** [comment]
Ah wait, the avatar-chaining is incorrect. Be right back with a fix...

**Comment by infinite-persistence** [comment]
Fixed, ready for review

File: app/scenes/Document/components/CommentThread.tsx [hunk 1]
```diff
@@ -192,6 +214,19 @@ function CommentThread({
       onClick={handleClickThread}
     >
       {commentsInThread.map((comment, index) => {
+        if (collapse !== null) {
+          if (index === collapse.begin) {
+            const count = collapse.final - collapse.begin + 1;
+            return (
+              <ShowMore onClick={handleClickExpand} key="show-more">
+                {t("Show {{ count }} reply", { count })}
```
Comment (infinite-persistence) [inline app/scenes/Document/components/CommentThread.tsx:222]: Hm ... not show why it isn't pluralized. Doc says just use `count` as the variable, right?

Comment (tommoor) [reply app/scenes/Document/components/CommentThread.tsx:222]: That's right, but the english translations need adjusting manually in this case.

**Comment by tommoor** [comment]
All seems good, I think the styling will need tweaking though – I want to keep the aesthetic that each thread is a "card" and the way this looks right now it feels like the card has a big gap in it with the background showing through. I can probably provide some designs in a few datys

**Comment by infinite-persistence** [comment]
Updated design

image

File: app/scenes/Document/components/CommentThread.tsx (hunk 1 above)
Comment (infinite-persistence) [reply app/scenes/Document/components/CommentThread.tsx]: Mybad. I just discovered `build:i18n`. It works now :smile:

**Comment by tommoor** [comment]
IMG_1692

How about like this? Having the `<Facepile />` on there is quite useful as you get a preview of who is hidden. It is similar to the first one but the difference in background would be much more subtle.

**Comment by tommoor** [comment]
Nice, thanks! I might want to tweak the background color a bit but probably easiest to do that post-merge

**Review by tommoor — APPROVED** [review]

**Pull request merged**